packages/gatsby-source-shopify/src/process-bulk-results.ts

```diff
@@ -0,0 +1,108 @@
+import { SourceNodesArgs } from "gatsby"
+
+import { shopifyTypes } from "./shopify-types"
+import {
+  isShopifyId,
+  createNodeId,
+  parseShopifyId,
+  decorateBulkObject,
+  processShopifyImages,
+} from "./helpers"
+
+interface IDecoratedResultChildren {
+  [key: string]: Array<string | IDecoratedResult>
+}
+
+export async function processBulkResults(
+  gatsbyApi: SourceNodesArgs,
+  pluginOptions: IShopifyPluginOptions,
+  results: BulkResults
+): Promise<number> {
+  let nodeCount = 0
+  const promises: Array<Promise<void>> = []
+  const children: IDecoratedResultChildren = {}
+
+  for (let i = results.length - 1; i >= 0; i -= 1) {
+    const result = decorateBulkObject(results[i]) as IDecoratedResult
+    const resultIsNode = Boolean(
+      result.shopifyId &&
+        JSON.stringify(Object.keys(result)) !== `["__parentId","shopifyId"]`
+    )
+
+    if (resultIsNode) {
+      const type = parseShopifyId(result.shopifyId)[1]
+      const imageFields = shopifyTypes[type].imageFields
+      const referenceFields = shopifyTypes[type].referenceFields
+
+      // Only increment for parent nodes
+      if (!result.__parentId) nodeCount++
+
+      // Attach reference fields
+      if (referenceFields) {
+        for (const referenceField of referenceFields) {
+          result[referenceField] = []
+        }
+      }
+
+      // Attach children references / objects
+      if (children[result.shopifyId]) {
+        for (const child of children[result.shopifyId]) {
+          const childType =
+            typeof child === `string`
+              ? parseShopifyId(child)[1]
+              : (child.__typename as string)
+          const field = shopifyTypes[childType].key
+
+          if (field) {
+            result[field] = [
+              typeof child === `string`
+                ? createNodeId(child, gatsbyApi, pluginOptions)
+                : child,
+              ...((result[field] || []) as Array<unknown>),
+            ]
+          } else {
+            throw new Error(
+              `There is no shopifyType for child type: ${childType}`
+            )
+          }
+        }
+      }
+
+      if (!isShopifyId(result.shopifyId)) {
+        throw new Error(
+          `Expected an ID in the format gid://shopify/<typename>/<id>`
+        )
+      }
+
+      const node = {
+        ...result,
+        id: createNodeId(result.shopifyId, gatsbyApi, pluginOptions),
+        internal: {
+          type: `${pluginOptions.typePrefix || ``}Shopify${type}`,
```

Comment:
```suggestion
          type: `${pluginOptions.typePrefix}Shopify${type}`,
```

Through the plugin options schema this will always be an empty string by default

Reply:
Removed this and the default for shopifyConnections across the package